File: bench/cases/map-str-100k/main.ts
const m = new Map<string, number>()
const n = 100000
for (let i = 0; i < n; i = i + 1) {
  const j = i % 4096
  const key = i % 2 === 0 ? "key" + j : "ключ" + j
  if (m.has(key)) {
    m.set(key, m.get(key)! + i)
  } else {
    m.set(key, i)
  }
}
let total = 0
for (const v of m.values()) {
  total = total + v
}
console.log(m.size, total)

File: bench/cases/map-str-100k/main.rs
use std::collections::HashMap;

fn main() {
    let mut m: HashMap<String, i64> = HashMap::new();
    let n: i64 = 100_000;
    for i in 0..n {
        let j = i % 4096;
        let key = if i % 2 == 0 {
            format!("key{}", j)
        } else {
            format!("ключ{}", j)
        };
        *m.entry(key).or_insert(0) += i;
    }
    let mut total: i64 = 0;
    for v in m.values() {
        total += v;
    }
    println!("{} {}", m.len(), total);
}
